eduhelx_jupyterlab_student/handlers.py

```diff
@@ -281,48 +186,188 @@ async def post(self):
             return
 
         current_assignment_path = student_repo.get_assignment_path(student_repo.current_assignment)
-        stage_files(".", path=student_repo.repo_root)
-        commit_id = commit(
-            submission_summary,
-            submission_description if submission_description else None,
-            path=student_repo.repo_root
-        )
-        push("origin", "master", path=student_repo.repo_root)
+        
+        rollback_id = get_head_commit_id(path=student_repo.repo_root)
+        stage_files(".", path=current_assignment_path)
+        
+        try:
+            commit_id = commit(
+                submission_summary,
+                submission_description if submission_description else None,
+                path=current_assignment_path
+            )
+        except Exception as e:
+            # If the commit fails then unstage the assignment files.
+            reset(".", path=current_assignment_path)
+            self.set_status(500)
+            self.finish(str(e))
+            return
+
         try:
             await self.api.create_submission(
                 student_repo.current_assignment["id"],
                 commit_id
             )
-            self.finish()
-        except requests.exceptions.HTTPError as e:
+        except Exception as e:
+            # If the submission fails create in the API, rollback the local commit to the previous head.
+            reset(rollback_id, path=student_repo.repo_root)
```

Comment:
it can be renamed to git_reset. Most of the git commands are just imported directly instead of aliased, should probably just be importing the git module directly to be hoenst.